packages/@aws-cdk/aws-appsync/lib/schema-base.ts

```diff
@@ -179,9 +218,28 @@ export class Directive {
   /**
    * the directive statement
    */
-  public readonly statement: string;
+  private statement: string;
 
-  private constructor(statement: string) { this.statement = statement; }
+  private readonly mode?: AuthorizationType;
+
+  private constructor(statement: string, mode?: AuthorizationType) {
+    this.statement = statement;
+    this.mode = mode;
+  }
+
+  /**
+   * Generate the directive statement
+   * @param modes the authorization modes of the graphql api
+   */
+  public toString(modes?: AuthorizationType[]): string {
+    if (modes && this.mode && !modes.some((mode) => mode === this.mode)) {
+      throw new Error(`No Authoration Type ${this.mode} declared in GraphQL Api.`);
+    }
+    if (this.mode === AuthorizationType.USER_POOL && modes && modes.length > 1) {
```

Comment:
does the following work?

```suggestion
    if (this.mode === AuthorizationType.USER_POOL && modes?.length > 1) {
```

Reply:
:( no